Rubberduck.Parsing/Symbols/ModuleBodyElementDeclaration.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Antlr4.Runtime;
+using Rubberduck.Parsing.Annotations;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Parsing.Symbols
+{
+    public abstract class ModuleBodyElementDeclaration : Declaration, IParameterizedDeclaration, ICanBeInterfaceMember, ICanBeDefaultMember
+    {
+        protected ModuleBodyElementDeclaration(
+            QualifiedMemberName name,
+            Declaration parent,
+            Declaration parentScope,
+            string asTypeName,
+            VBAParser.AsTypeClauseContext asTypeContext,
+            string typeHint,
+            Accessibility accessibility,
+            DeclarationType type,
+            ParserRuleContext context,
+            ParserRuleContext attributesPassContext,
+            Selection selection,
+            bool isArray,
+            bool isUserDefined,
+            IEnumerable<IAnnotation> annotations,
+            Attributes attributes)
+            : base(
+                name,
+                parent,
+                parentScope,
+                asTypeName,
+                typeHint,
+                false,
+                false,
+                accessibility,
+                type,
+                context,
+                attributesPassContext,
+                selection,
+                isArray,
+                asTypeContext,
+                isUserDefined,
+                annotations,
+                attributes)
+        {
+            if (!(parent is ModuleDeclaration module))
+            {
+                return;
+            }
+
+            module.AddMember(this);
+
+            _interface = new Lazy<(bool IsInterfaceImplementation, ClassModuleDeclaration InterfaceImplemented)>(() => ResolveInterface(this), true);
+            _implemented = new Lazy<Declaration>(() => MemberImplemented(this), true);
+        }
+
+        /// <inheritdoc/>
+        public bool IsDefaultMember => this.IsDefaultMember();
+
+        private readonly List<ParameterDeclaration> _parameters = new List<ParameterDeclaration>();
+        public IEnumerable<ParameterDeclaration> Parameters => _parameters.ToList();
+
+        public void AddParameter(ParameterDeclaration parameter)
```

Comment:
why is this public, shouldn't it be `internal`? Same justification as for `AddMembers`

Reply:
Found it that way.  It's also member of `IParameterizedDeclaration`, so it currently has to be public because that interface is used outside of `Rubberduck.Parsing`.